src/main/java/teammates/ui/webapi/JoinCourseAction.java

```diff
@@ -72,6 +75,8 @@ private JsonResult joinCourseForInstructor(String regkey) throws InvalidOperatio
             throw new EntityNotFoundException(ednee);
         } catch (EntityAlreadyExistsException eaee) {
             throw new InvalidOperationException(eaee);
+        } catch (JoinCourseException jce) {
+            throw new InvalidOperationException(jce.getMessage());
```

Comment:
I haven't taken a look at the rest of the PR, but throwing like this is generally not good practice because it does not preserve stack trace. Pass the throwable as the argument instead of the message.

Reply:
Ok thanks, I made the changes